vllm/spec_decode/top1_proposer.py

```diff
@@ -226,47 +226,31 @@ def _merge_outputs(
         if maybe_sampler_output is None:
             # If no speculative tokens, the sampler output will be None.
             # In this case we return empty proposals.
-            proposal_tokens = torch.tensor(-1,
-                                           dtype=torch.long,
-                                           device=self._device).expand(
-                                               batch_size, proposal_len)
-            proposal_probs = torch.tensor(0,
-                                          dtype=torch.float32,
-                                          device=self._device).expand(
-                                              batch_size, proposal_len,
-                                              self._vocab_size)
-            proposal_lens_tensor = torch.tensor(0,
-                                                dtype=torch.long,
-                                                device=self._device).expand(
-                                                    len(proposal_lens))
+            proposal_tokens = torch.full((batch_size, proposal_len), -1,
+                                         dtype=torch.long, device=self._device)
+            proposal_probs = torch.zeros((batch_size, proposal_len, self._vocab_size),
+                                         dtype=torch.float32, device=self._device)
+            proposal_lens_tensor = torch.zeros(len(proposal_lens),
+                                               dtype=torch.long, device=self._device)
```

Comment:
What's the reason for this change? the existing code minimizes the allocations, I'd be surprised if this was faster.

Reply:
I confirmed that the speed decrease occurs due to `cudamemcpy` in the process of generating tensor.
As you can see, the PR code has the same result, and you can get the effect of speed improvement.

```
if maybe_sampler_output is None:
    with nvtx.annotate("allocation"):
```
I set up nvtx as above and checked the profiling result below.
Above is the previous code, below is the PR code.
<img width="1618" alt="2024-08-27_16-18-05" src="https://github.com/user-attachments/assets/5f4d3937-684f-4424-89b8-fb2b5ca2791b">